### ISSUE TITLE
Get a scalar after aggregation from a dataset

### ISSUE THREAD

**Issue by antonio-olleros** [sdmx-twg/vtl#384]
### Issue Description
At the moment it is not possible to extract a scalar from a dataset.
Suppose that you want to count() the number of records of a dataset (e.g. `number := count(DS)`). What you get as a result is a dataset with no indentifiers and a single measure var_int.
Suppose now that you want to use that result to filter some other dataset. At the moment it is not possible to do that in VTL.
A possible workaround would be to do a join between the datasets, but an inner join would produce no records by definition. So we would need to do a cross join and only then we could do the filter, which is complex and misleading.

### Proposed Solution
My favourite solution would be changing the especification for the membership operator `#` specifying that for a dataset with no identifiers, the operator returns a scalar.

**Comment by NicoLaval** [comment]
Interesting topic.
I remember when I was still at INSEE that it was a need in the collection channel to get a particular datapoint.

But, as VTL has not index for lines, it didn't seem possible to do something like `get(1, me_1)`.

In your case, when dataset contains 1 column and 1 line, the scalar identification is easy, but I think using membership with an exception in this case is not really beautiful (membership would generally return a dataset with a reduced number of columns, but in an other case a scalar, it's not consistent).

It would maybe better to create a new operator, dedicated to this use case, but it would be the only one which take a dataset as argument and returning a scalar. Do we really want that?

The join solution seems to be cleanest for me.

**Comment by antonio-olleros** [comment]
I think that the join solution is very problematic. Try to make a business user that to filter by a result they need  to write something like:

```
number_records := count(DS1);
joined_ds := cross_join(DS2, number_records);
filtered_ds := joined_ds[filter me1 > int_var];
```

The business is very clear, the VTL is complex and far from the business need... For me not using membership is fine, but at least we should create a new operator, so that we could write something like:

```
result := DS2[filter me1 > get_scalar(count(DS1), int_var)];
```
It is representing much more effectively the transformation

**Comment by NicoLaval** [comment]
Agree it's more intuitive that way.

But is this "reducing" dimension operator (ds to scalar) consistent with VTL philosophy?

If so, why not

**Comment by linardian** [comment]
Allowing cubes (dataset) without dimensions (i.e. scalar) is a very controversial issue even for SDMX, long discusse during some TWG meetings. I am very skeptic in introducing it (I have been against also during TWG discussions) and at the moment this new "item" has been "frozen" for SDMX. But ready and glad to continue discussing in next VTL meeting, if you deem of interest.

**Comment by antonio-olleros** [comment]
Hi Angelo!
Datasets without identifiers (dimensions) are already allowed, and scalars too. But the problem is that a dataset without identifiers and with one single measure are not equivalent to scalars, because, for instance, it is not possible to use it inside a calc

**Comment by linardian** [comment]
Of course, Antonio, dataset without identifiers and scalar are not equivalent, but this topic has the same "background" as the one discussed in the TWG, in a nutshell promoting a scalar to a ds without dimensions in order to put it in relation with other "real" datasets. That said I have to deeply analyse the # proposal and other possible "easy workarounds" to implement the use case you provide.

**Comment by antonio-olleros** [comment]
As discussed in the VTL TF Meeting in Frankfurt, I'm proposing a new operator here:
https://github.com/sdmx-twg/vtl/tree/Feat/get-scalar/v2.2/docs/reference_manual/operators/General%20purpose%20operators/Get%20scalar

**Comment by linardian** [comment]
As discussed in the VTL TF Meeting in Frankfurt, I have "extended" the membership operator (#), modifying the behaviour text and adding Antonio's examples from get_scalar (as ex_7 and ex_8). Please @antonio-olleros, double check and if you like you can drop your get-scalar branch.

**Comment by antonio-olleros** [comment]
Hi @linardian , this is fine with me!

**Comment by NicoLaval** [comment]
Hi @antonio-olleros @linardian,
What troubles me is the fact that a single symbol produces 2 completely different operations.
Select a Component or reduce a dataset to a scalar is not the same concept. That's why in think we have to provide a dedicated notation/operator to avoid confusion.
This will cause the end user to misunderstand what `#` produces in one context and not in another (not for VTL experts but for basic users).
But if the majority is ok, let's go, no veto.
@linardian what was your opinion?

**Comment by linardian** [comment]
Hi @NicoLaval 

In principle I agree with you (also because the same symbol is used for a completely different purpose in join operations).
But logically it is just the application of the same transformation (from dataset to dataset if an Id is provided, from dataset to component if a Measure/Attribute is provided) for a special case of a dataset without identifiers (and obviously with just one instance). For me defining a new operator usable for only a very special case is not worth. Obviously, if there will be some complaints from (basic) users we can always change our mind. Glad to receive more feedbacks.
